tests/unit/components/test_blockutils.py

```diff
@@ -56,29 +56,35 @@ def pretend_blocklists(tmpdir):
     return urls, bl_dst_dir
 
 
-def test_blocklist_dl(pretend_blocklists):
-    num_single = 0
+def test_blocklist_dl(qtbot, pretend_blocklists):
+    total_expected = 10
+    num_single_dl_called = 0
+    all_dl_called = False
+
+    def on_single_download(download: IO[bytes]) -> None:
+        nonlocal num_single_dl_called
+        num_single_dl_called += 1
 
-    def on_single_download(download: typing.IO[bytes]) -> None:
-        nonlocal num_single
-        num_single += 1
         num_lines = 0
         for line in io.TextIOWrapper(download, encoding="utf-8"):
             assert line.split(".")[-1].strip() in ("com", "net", "is")
             num_lines += 1
         assert num_lines >= 1
 
     def on_all_downloaded(done_count: int) -> None:
-        assert done_count == 10
+        nonlocal all_dl_called
+        assert done_count == total_expected
+        all_dl_called = True
```

Comment:
I think you can drop this entirely - `qtbot.waitSignal` will already raise a `TimeoutError` if the signal doesn't get called, and we're already checking the signals argument via `blocker.args`.